### PR TITLE
Extend Yola client with new WLTaskResourceMixin

### LINKED ISSUE #80: Add WL task resource and use it in the client
### Summary
In the scope of in-app purchases implementation for Bottomline, we need to extend `yolapy` to query [new wl-task endpoints](https://github.com/yola/wlproxy/blob/b0ec575cd758a1c6a95a5b5a9e621d937f2c5513/wlproxy/wl_tasks/views.py#L54-L68). 

### Todo
- add `WLTaskResourceMixin` (see exisiting mixing from `yolapy/resources/*.py` for inspiration)
  - the mixin should have methods to query [wl-task endpoints](https://github.com/yola/wlproxy/blob/b0ec575cd758a1c6a95a5b5a9e621d937f2c5513/wlproxy/wl_tasks/views.py#L54-L68) (follow the link). 
- use the new mixin in [the HTTP client](https://github.com/yola/yolapy/blob/f7b43ba28fb81607c9c6094234aae00dd3c86c03/yolapy/services.py#L7).

### PR DESCRIPTION
**Pull request by serhii-karelov**
Closes https://github.com/yola/yolapy/issues/80

Summary:
* Add `resources.wl_task.WLTaskResourceMixin` with the following methods:
  * `complete_wl_task(wl_task_id)`
  * `fail_wl_task(wl_task_id, reason)`
* Extend `Yola` with the new `WLTaskResourceMixin`.

Note:
There are integration tests. Unfortunately, we do not have a partner endpoint to create a WL Task. Thus it makes hard to test thing in integration. That's why integration tests for the new resource have not been added.

### CODE REVIEW

File: yolapy/resources/wl_task.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+class WLTaskResourceMixin(object):
+    """Methods for managing WL Task resources."""
+
+    def complete_wl_task(self, wl_task_id):
```
Comment (snitch) [inline yolapy/resources/wl_task.py:4]: D202 No blank lines allowed after function docstring

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: .travis.yml [hunk 2]
```diff
@@ -2,9 +2,10 @@ sudo: false
 language: python
 python:
   - "2.7"
-  - "3.3"
-  - "3.4"
   - "3.5"
+  - "3.6"
```
Comment (serhii-karelov) [inline .travis.yml:6]: Updated according to https://devguide.python.org/#branchstatus

Thanks @kosmos342 for suggestion and examples from https://github.com/yola/demands/pull/76/files

Comment (serhii-karelov) [reply .travis.yml:6]: Also, https://www.python.org/downloads/release/python-3410/ says that "Python 3.4 has reached end-of-life".

File: CHANGELOG.rst [hunk 3]
```diff
@@ -1,3 +1,10 @@
+0.6.2
+------------------
+* Add `resources.wl_task.WLTaskResourceMixin` with the following methods:
+  * `complete_wl_task(wl_task_id)`
+  * `complete_wl_task(wl_task_id)`
```
Comment (kosmos342) [inline CHANGELOG.rst]: copy-paste?)

Comment (serhii-karelov) [reply CHANGELOG.rst]: +
fixing

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: yolapy/services.py [hunk 4]
```diff
@@ -1,14 +1,24 @@
...
+)
 
 
 class Yola(
         HTTPServiceClient, campaign.CampaignResourceMixin,
         cname_zone.CnameZoneMixin, partner.PartnerResourceMixin,
         site.SiteResourceMixin, siteimport.SiteImportResourceMixin,
-        subscription.SubscriptionResourceMixin, user.UserResourceMixin):
+        subscription.SubscriptionResourceMixin, user.UserResourceMixin,
+        wl_task.WLTaskResourceMixin):
```
Comment (kosmos342) [inline yolapy/services.py]: What do you think about every mixin in a separate line?

Comment (serhii-karelov) [reply yolapy/services.py]: > What do you think about every mixin in a separate line?

👍 I actually like such style more and more than happy to improve it!

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: yolapy/services.py [hunk 5]
```diff
@@ -1,14 +1,29 @@
...
-        subscription.SubscriptionResourceMixin, user.UserResourceMixin):
+    HTTPServiceClient,
+    campaign.CampaignResourceMixin,
+    cname_zone.CnameZoneMixin,
+    partner.PartnerResourceMixin,
+    site.SiteResourceMixin,
+    siteimport.SiteImportResourceMixin,
+    subscription.SubscriptionResourceMixin,
+    user.UserResourceMixin,
+    wl_task.WLTaskResourceMixin
```
Comment (kosmos342) [inline yolapy/services.py]: what do you think about trailing comma?) it can simplify changelog of future PRs

File: setup.py [hunk 6]
```diff
@@ -29,6 +29,7 @@
     install_requires=[
         'demands >= 4.0.0, < 5.0.0',
     ],
+    python_requires = ['==2.7, ~=3.5'],
```
Comment (snitch) [inline setup.py]: E251 unexpected spaces around keyword / parameter equals

File: yolapy/services.py (hunk 5 above)
Comment (serhii-karelov) [reply yolapy/services.py]: I think that trailing commas are good in such cases. +1

File: setup.py [hunk 7]
```diff
@@ -29,6 +29,7 @@
     install_requires=[
         'demands >= 4.0.0, < 5.0.0',
     ],
+    python_requires ='==2.7, ~=3.5',
```
Comment (snitch) [inline setup.py]: E251 unexpected spaces around keyword / parameter equals

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Review by kosmos342 — APPROVED** [review]

**Pull request merged**